nobl9/resource_alertmethod.go

```diff
@@ -224,6 +257,17 @@ func (i alertMethodWebhook) UnmarshalSpec(d *schema.ResourceData, spec v1alphaAl
 	err = d.Set("template_fields", config.TemplateFields)
 	diags = appendError(diags, err)
 
+	headers := make(map[string]interface{})
+
+	for _, header := range config.Headers {
+		if !header.IsSecret { // secrets will be returned from the API as [hidden] values
+			headers[header.Name] = header.Value
+		}
+	}
+
+	err = d.Set("headers", headers)
+	diags = appendError(diags, err)
```

Comment:
don't you need to read the sensitive ones as well here? won't it mess up the plan if they're gone? just wondering

Reply:
actually if I do it will mess up the plan. What would I pass to the `sensitive_headers`? empty map? map with keys but empty values? either would cause a diff.

The tests show that after applying and getting again, there is no diff. I know it's a bit counter intuitive but works.
When I add an empty map in the unmarshall (and made it sensitive: false to see it better) I get: 
![image](https://github.com/user-attachments/assets/646568d5-6513-44b0-a108-a7e653f017a6)

Reply:
huh, good to know :) thanks!